File: e2e-pytest-playwright/pages/flight_page.py
from pages.abstract_page import AbstractPage

class FlightPage(AbstractPage):

  def select_trip_type(self, trip_type):
    elem = self.driver.locator("//input[@name='tripType' and @value='" + trip_type + "']")
    elem.click()

  def select_depart_from(self, from_port):
    select_elem = self.driver.locator("//select[@name='fromPort']")
    select_elem.select_option(from_port)

  def select_arrive_at(self, to_port):
    select_elem = self.driver.locator("//select[@name='toPort']")
    select_elem.select_option(to_port)

  def select_depart_day(self, depart_day):
    select_elem = self.driver.locator("//select[@name='departDay']")
    select_elem.select_option(depart_day)

  def select_depart_month(self, depart_month):
    select_elem = self.driver.locator("#departMonth")
    select_elem.select_option(depart_month)

  def select_return_day(self, return_day):
    Select(self.driver.find_element_by_name("returnDay")).select_by_visible_text(return_day)

  def select_return_month(self, return_month):
    Select(self.driver.find_element_by_id("returnMonth")).select_by_visible_text(return_month)

  def click_continue(self):
    self.driver.locator("//input[@value='Continue']").click()

File: e2e-junit-selenium/src/pages/FlightPage.java
package pages;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;

public class FlightPage < AbstractPage {

    public FlightPage(WebDriver aDriver) {
        super(aDriver);
    }
    

    public void selectTripType(String trip_type) {
      driver.findElement(By.xpath("//input[@name='tripType' and @value='" + trip_type + "']").click();
    }

    public void selectDepartFrom(String from_port) {
      Selenium::WebDriver::Support::Select.new(driver.find_element(:name, "fromPort")).select_by(:text, from_port)
    }

    public void selectArriveAt(String to_port) {
      Selenium::WebDriver::Support::Select.new(driver.find_element(:name, "toPort")).select_by(:text, to_port)
    }

    public void selectDepartDay(String depart_day) {
      Selenium::WebDriver::Support::Select.new(driver.find_element(:id, "departDay")).select_by(:text, depart_day)
    }

    public void selectDepartMonth(String depart_month) {
      Selenium::WebDriver::Support::Select.new(driver.find_element(:id, "departMonth")).select_by(:text, depart_month)
    }

    public void select_return_day(return_day) {
      Selenium::WebDriver::Support::Select.new(driver.find_element(:id, "returnDay")).select_by(:text, return_day)
    }

    public void select_return_month(return_month) {
      Selenium::WebDriver::Support::Select.new(driver.find_element(:id, "returnMonth")).select_by(:text, return_month)
    }
    
    public void clickContinue() {
      driver.findElement(By.xpath(,"//input[@value='Continue']").click():
    }
                      
  }
